src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -627,21 +561,28 @@ public static ClassType GetClassType(Type type, bool checkForConverter, JsonSeri
             Debug.Assert(type != null);
 
             runtimeType = type;
+
             nullableUnderlyingType = Nullable.GetUnderlyingType(type);
 
+            // Type is nullable e.g. typeof(int?).
             if (nullableUnderlyingType != null)
             {
+                // Check if there's a converter for this nullable type, e.g. do we have a converter that implements
+                // JsonConverter<int?> if the type is typeof(int?)?
                 converter = options.DetermineConverterForProperty(parentClassType, type, propertyInfo);
 
                 if (converter == null)
                 {
+                    // No converter. We'll check below if there's a coonverter for the non-nullable type e.g.
+                    // one that implements JsonConverter<int>, given the type is typeof(int?).
                     type = nullableUnderlyingType;
-                    parentClassType = type;
+                    parentClassType = nullableUnderlyingType;
```

Comment:
Is parentClassType just a dummy here? Why is it needed?

Reply:
It's a dummy here. I'll remove this line.